explainaboard/metric.py

```diff
@@ -418,12 +418,21 @@ class QAMetric(Metric):
 
     def normalize_answer(self, s: str) -> str:
         """Lower text and remove punctuation, articles and extra whitespace."""
-        s = re.sub(r'\b(a|an|the)\b', ' ', s)
-        s = ' '.join(s.split())
-        exclude_punc = set(string.punctuation)
-        s = ''.join(ch for ch in s if ch not in exclude_punc)
-        s = s.lower()
-        return s
+
```

Comment:
Sorry, I realize my bug now. I'd suggest the following code, which should be equivalent to the previous code but less verbose:

        s = s.lower()
        exclude_punc = set(string.punctuation)
        s = ''.join(ch for ch in s if ch not in exclude_punc)
        s = re.sub(r'\b(a|an|the)\b', ' ', s)
        s = ' '.join(s.split())
        return s

Reply:
Hey, Graham, thanks!  I just introduced the concept of language to the Metric class.  So far, I followed the official script (from MLQA) Jinlan shared with me (also a little verbose). You probably could take a look.

Reply:
The following look a bit specific to the MLQA dataset and seems like maybe we shouldn't have it directly in the top-level QAMetric class?

As an alternative suggestion, maybe we could have an analogous class to `Tokenizer` that is something like `Preprocessor`, and we could have `MLQAPreprocessor`. Then `QAMetric` could take a `Preprocessor` as an argument, and apply the appropriate processing in a more generic way before doing actual evaluation.